lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -58,6 +58,10 @@ def self.validate_MSFT_Credential(name, value)
         unless extraneous.empty?
           fail "#{name} includes invalid keys: #{extraneous.join(',')}"
         end
+
+        if value.values.any?{|v| v.nil? || v.length == 0}
```

Comment:
This doesn't tell the user which value is empty, just that there is something empty. Is there a better way that isn't hardcoded?